chapters/extend/executor.rst

```diff
@@ -221,11 +257,154 @@ Then simply use it in Jina CLI by specifying ``jina pod --uses=my.yml``, or ``Fl
+            tensor = clip.tokenize(data)
+            with torch.no_grad():
+                encoded_data = self.model.encode_text(tensor)
+            return encoded_data.cpu().numpy()
+
+In the code sample above, we called CLIP's :meth:`encode_text` to use the pre-trained CLIP model and encode input data into vector embeddings.
+
+.. note:: The example above is a minimum working example of a `CLIPTextEncoder`, for full features such as GPU support, batching and dockerization, please checkout `Jina-hub <https://github.com/jina-ai/jina-hub/tree/master/encoders>`_.
+
+The same applies to `CLIPImageEncoder`, the only difference is to use :meth:`self.model.encode_image` in :meth:`encode`.
+Last but not least, create the YAML configuration for the encoder and use it with Jina CLI or Flow API.
+
+.. highlight:: yaml
+.. code-block:: yaml
+
+    !CLIPTextEncoder
+    metas:
+      py_modules:
+        - __init__.py
+
+Then use it in Jina CLI by specifying ``jina pod --uses=config.yml``,
+or ``Flow().add(uses='config.yml')`` in Flow API.
+And you have a good foundation to build your index/query Flow powered by CLIP.
+
+Share Your Work!
+^^^^^^^^^^^^^^^^^^^^^
+
+If you would like to share your customized Executor with the community, more than welcome!
+We use `cookiecutter <https://github.com/cookiecutter/cookiecutter>`_ to create Jina Executor from the template.
+
+.. note:: Install Docker and run `pip install "jina[devel]"` before you start.
+
+To make sure your work has a good shape, Jina provides a wizard to help you create a Executor, start it with `jina hub new --type pod`.
+It will generate a standard Executor project like this:
+
+.. highlight:: text
+.. code-block:: text
+
+    CLIPTextEncoder/
+    ├── Dockerfile
+    ├── manifest.yml
+    ├── README.md
+    ├── config.yml
+    ├── requirements.txt
+    ├── __init__.py
+    └── tests/
+        ├── test_CLIPTextEncoder.py
+        └── __init__.py
+
+And you can put your customized Encoder, such as `CLIPTextEncoder` inside `__init__.py`.
+The YAML configurations should be placed in `config.yml`.
+
+To ensure your customised Executor, such as `CLIPTextEncoder` performs exactly the same as the original CLIP model,
+please add tests inside `tests` folder.
+For example, encode some text data with the raw CLIP model, and assert we get the same result with `CLIPTextEncoder`.
+
+Please build and test your Encoder locally with:
+
+.. highlight:: shell
+.. code-block:: shell
+
+    jina hub build -t jinahub/type.kind.jina-image-name:image_version-jina_version <your_folder>
+
+Once tested, you should login to jina hub with `jina hub login`, copy/paste the token into GitHub to verify your account.
+Now you are able to push your work to jina hub:
+
+.. highlight:: shell
+.. code-block:: shell
+
+    jina hub push jinahub/type.kind.jina-image-name:image-jina_version
+
+In our example, the type is `pod`, kind is `encoders` and `jina-image-name` is `cliptextencoder` and `clipimageencoder`.
```

Comment:
I do not understand this sentence very well. The word `kind ` should be there? what do you mean?

Reply:
@davidbp see the above line of code of the docker naming convention:
```
jina hub push jinahub/type.kind.jina-image-name:image-jina_version
```